libvips/foreign/vips2jpeg.c

```diff
@@ -627,17 +628,24 @@ write_vips( Write *write, int qfac, const char *profile,
 	if( progressive ) 
 		jpeg_simple_progression( &write->cinfo ); 
 
-	/* Turn off chroma subsampling. Follow IM and do it automatically for
-	 * high Q. 
-	 */
-	if( no_subsample ||
-		qfac >= 90 ) { 
-		int i;
-
-		for( i = 0; i < in->Bands; i++ ) { 
-			write->cinfo.comp_info[i].h_samp_factor = 1;
-			write->cinfo.comp_info[i].v_samp_factor = 1;
-		}
+	switch (subsample_mode) {
```

Comment:
libvips indentation style is K&R, so:

```C
switch( subsample_mode ) {
case VIPS_...ON:
        break;
```

ie. hard tabs set to 8, 80 char line length.

Reply:
done